src/detect-engine-threshold.c

```diff
@@ -109,9 +127,56 @@ static int ThresholdEntrySet(void *dst, void *src)
     return 0;
 }
 
+static void ThresholdDistinctInit(ThresholdEntry *te, const DetectThresholdData *td)
+{
+    if (td->type != TYPE_DETECTION || td->unique_on == DF_UNIQUE_NONE || td->seconds == 0) {
+        return;
+    }
+
+    const uint32_t bitmap_size = DF_PORT_BITMAP_SIZE;
+    te->distinct_seen = 0;
+#ifdef UNITTESTS
+    if (g_threshold_force_alloc_fail) {
+        te->distinct_bitmap_union = NULL;
+        return;
+    }
+#endif
+    te->distinct_bitmap_union = SCCalloc(1, bitmap_size);
```

Comment:
I would like to be able to observe this condition, best by a counter I think

Additionally, I would like to know how these allocations are bounded. The creation of a `ThresholdEntry` is based on src_ip in most cases, so an attacker could trigger the creation of many trackers, each having a 8KiB allocation?